src/create-model.jl

```diff
@@ -1034,20 +1034,19 @@ function create_model(
             ]
         @expression(
             model,
-            accumulated_units[
-                y ∈ Y,
-                a ∈ decommissionable_assets_using_simple_method∪decommissionable_assets_using_compact_method,
-            ],
+            accumulated_units[y ∈ Y, a ∈ A],
             if a in decommissionable_assets_using_simple_method
                 accumulated_units_simple_method[y, a]
-            else
+            elseif a in decommissionable_assets_using_compact_method
                 sum(
                     accumulated_units_compact_method[accumulated_set_using_compact_method_lookup[(
                         a,
                         y,
                         v,
                     )]] for v in V_all if (a, y, v) in accumulated_set_using_compact_method
                 )
+            else
+                sum(values(graph[a].initial_units[y]))
             end
         )
```

Comment:
My earlier comment was to transform this into the following where each condition has its own expression.

```suggestion
  accumulated_units = model[:accumulated_units] = [
    if a in decommissionable_assets_using_simple_method
      @expression(model,
        accumulated_units_simple_method[y, a]
      )
    elseif a in decommissionable_assets_using_compact_method
      @expression(model,
        sum(
          accumulated_units_compact_method[accumulated_set_using_compact_method_lookup[(
            a,
            y,
            v,
          )]] for v in V_all if (a, y, v) in accumulated_set_using_compact_method
        )
      )
    else
      @expression(model,
        sum(values(graph[a].initial_units[y]))
      )
    end
    for y ∈ Y, a ∈ A
  ]
```